Makefile.rules

```diff
@@ -147,15 +147,21 @@ GAP_CPPFLAGS += $(BOEHM_GC_CPPFLAGS)
 GAP_CPPFLAGS += $(LIBATOMIC_OPS_CPPFLAGS)
 
 # Enable warnings
-GAP_CPPFLAGS += -Wall -Wextra \
+GAP_CPPFLAGS += \
+	-Wall \
+	-Wextra \
+	-pedantic \
+	-Wshadow
+
+# ... disable some annoying warnings ...
+GAP_CPPFLAGS += \
 	-Wno-unused-parameter \
 	-Wno-sign-compare \
 	-Wno-implicit-fallthrough
 
 # ... and disable warnings about unknown warnings ...
-GAP_CPPFLAGS += \
-	-Wno-unknown-warning \
-	-Wno-unknown-warning-option
+GAP_CPPFLAGS += -Wno-unknown-warning			# ... for gcc
```

Comment:
While this was already a bug, I'm fairly positive `-Wno-unknown-warning` isn't a gcc option.